### PR TITLE
Add native xcode 13 retries

### PR DESCRIPTION
**Pull request by joshdholtz**
### Motivation and Context
#19612

### Description
Use `-retry-tests-on-failure -test-iterations <number>` if `number_of_retries` >= 1 and using Xcode 13

### CODE REVIEW

File: scan/lib/scan/test_command_generator.rb [hunk 1]
```diff
@@ -72,6 +72,12 @@ def options # rubocop:disable Metrics/PerceivedComplexity
       end
       options << "-xctestrun '#{config[:xctestrun]}'" if config[:xctestrun]
       options << config[:xcargs] if config[:xcargs]
+      if config[:number_of_retries] >= 1
+        if FastlaneCore::Helper.xcode_at_least?(13)
```
Comment (lucgrabowski) [inline scan/lib/scan/test_command_generator.rb:76]: Those two conditions could be joined into one if statement.

File: scan/lib/scan/runner.rb [hunk 2]
```diff
@@ -51,7 +51,9 @@ def test_app
         end
       end
 
-      execute(retries: Scan.config[:number_of_retries])
+      # Set retries to 0 if Xcode 13 because TestCommandGenerator will set '-retry-tests-on-failure -test-iterations'
```
Comment (lucgrabowski) [inline scan/lib/scan/runner.rb:54]: Thanks for adding the explanation, it helped me understand the PR's code ;)

**Review by lucgrabowski — APPROVED** [review]
LGTM 🔥

File: scan/lib/scan/test_command_generator.rb (hunk 1 above)
Comment (joshdholtz) [reply scan/lib/scan/test_command_generator.rb:76]: Oh whoops! I was angry at the tests and split this up to debug and never put it back together 😛

**Review by piotrsmajek — APPROVED** [review]

**Pull request merged**